### ISSUE TITLE
instance-selector-cpu-architecture strange behavior...?

### ISSUE THREAD

**Issue by ecc256** [weaveworks/eksctl#4336]
Guys,
I'm seeing following behavior:
`--instance-selector-cpu-architecture x86_64` does select **arm64** image.
`--instance-selector-cpu-architecture arm64` cannot find any images.
Am I doing something wrong?

```
ami-03fd9958c029011df
ubuntu-eks/k8s_1.21/images/hvm-ssd/ubuntu-focal-20.04-arm64-server-20211005
```
```
ami-0e7eae3cdfae5d370
ubuntu-eks/k8s_1.21/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20211005
```
```
# eksctl create nodegroup \
>  --cluster k8s \
>  --region us-east-1 \
>  --name k8s \
>  --nodes 3 \
>  --nodes-min 3 \
>  --nodes-max 6 \
>  --node-volume-size 256 \
>  --node-volume-type gp3 \
>  --node-ami-family Ubuntu2004 \
>  --instance-selector-vcpus 2 \
>  --instance-selector-memory 8 \
>  --instance-selector-gpus 0 \
>  --instance-selector-cpu-architecture x86_64 \
>  --alb-ingress-access \
>  --ssh-access \
>  --ssh-public-key tnetadmin \
>  --subnet-ids subnet-6689612c,subnet-cd654290,subnet-74e6c75b \
>  --managed
```
2021-10-13 20:42:45 [ℹ]  eksctl version 0.69.0
2021-10-13 20:42:45 [ℹ]  using region us-east-1
2021-10-13 20:42:45 [ℹ]  will use version 1.21 for new nodegroup(s) based on control plane version
2021-10-13 20:42:47 [ℹ]  nodegroup "k8s" will use "**ami-03fd9958c029011df**" [Ubuntu2004/1.21]
2021-10-13 20:42:47 [ℹ]  using EC2 key pair %!q(*string=<nil>)
2021-10-13 20:42:48 [ℹ]  1 existing nodegroup(s) (k8s) will be excluded
2021-10-13 20:42:48 [ℹ]  2 sequential tasks: { fix cluster compatibility, no tasks }
2021-10-13 20:42:48 [ℹ]  checking cluster stack for missing resources
2021-10-13 20:42:48 [ℹ]  cluster stack has all required resources
2021-10-13 20:42:48 [ℹ]  no tasks
2021-10-13 20:42:48 [✔]  created 0 nodegroup(s) in cluster "k8s"
2021-10-13 20:42:48 [✔]  created 0 managed nodegroup(s) in cluster "k8s"
2021-10-13 20:42:49 [ℹ]  checking security group configuration for all nodegroups
2021-10-13 20:42:49 [ℹ]  all nodegroups have up-to-date configuration

```
# eksctl create nodegroup \
>  --cluster k8s \
>  --region us-east-1 \
>  --name k8s \
>  --nodes 3 \
>  --nodes-min 3 \
>  --nodes-max 6 \
>  --node-volume-size 256 \
>  --node-volume-type gp3 \
>  --node-ami-family Ubuntu2004 \
>  --instance-selector-cpu-architecture arm64 \
>  --instance-selector-vcpus 2 \
>  --instance-selector-memory 8 \
>  --instance-selector-gpus 0 \
>  --alb-ingress-access \
>  --ssh-access \
>  --ssh-public-key tnetadmin \
>  --subnet-ids subnet-6689612c,subnet-cd654290,subnet-74e6c75b \
>  --managed

```
2021-10-13 20:44:24 [ℹ]  eksctl version 0.69.0
2021-10-13 20:44:24 [ℹ]  using region us-east-1
2021-10-13 20:44:24 [ℹ]  will use version 1.21 for new nodegroup(s) based on control plane version
2021-10-13 20:44:27 [✖]  image family Ubuntu2004 doesn't support ARM image class
Error: unable to determine AMI to use: unable to determine AMI for region us-east-1, version 1.21, instance type m6g.large and image family Ubuntu2004

**Comment by ecc256** [comment]
One more:
```
# eksctl create nodegroup \
>  --cluster k8s \
>  --region us-east-1 \
>  --name k8sx \
>  --nodes 3 \
>  --nodes-min 3 \
>  --nodes-max 6 \
>  --node-volume-size 256 \
>  --node-volume-type gp3 \
>  --node-ami-family Ubuntu2004 \
>  --instance-selector-cpu-architecture x86_64 \
>  --instance-selector-vcpus 2 \
>  --instance-selector-memory 8 \
>  --instance-selector-gpus 0 \
>  --alb-ingress-access \
>  --ssh-access \
>  --ssh-public-key tnetadmin \
>  --subnet-ids subnet-6689612c,subnet-cd654290,subnet-74e6c75b \
>  --managed
```
2021-10-13 21:04:12 [ℹ]  eksctl version 0.69.0
2021-10-13 21:04:12 [ℹ]  using region us-east-1
2021-10-13 21:04:12 [ℹ]  will use version 1.21 for new nodegroup(s) based on control plane version
2021-10-13 21:04:14 [ℹ]  nodegroup "k8sx" will use "ami-03fd9958c029011df" [Ubuntu2004/1.21]
2021-10-13 21:04:15 [ℹ]  using EC2 key pair %!q(*string=<nil>)
2021-10-13 21:04:15 [ℹ]  1 existing nodegroup(s) (k8s) will be excluded
2021-10-13 21:04:15 [ℹ]  1 nodegroup (k8sx) was included (based on the include/exclude rules)
2021-10-13 21:04:15 [ℹ]  will create a CloudFormation stack for each of 1 managed nodegroups in cluster "k8s"
2021-10-13 21:04:15 [ℹ]  2 sequential tasks: { fix cluster compatibility, 1 task: { 1 task: { create managed nodegroup "k8sx" } } }
2021-10-13 21:04:15 [ℹ]  checking cluster stack for missing resources
2021-10-13 21:04:16 [ℹ]  cluster stack has all required resources
2021-10-13 21:04:16 [ℹ]  building managed nodegroup stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:04:16 [ℹ]  deploying stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:04:16 [ℹ]  waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:04:32 [ℹ]  waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:04:49 [ℹ]  waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:05:09 [ℹ]  waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:05:26 [ℹ]  waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:05:26 [✖]  unexpected status "ROLLBACK_COMPLETE" while waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx"
2021-10-13 21:05:26 [ℹ]  fetching stack events in attempt to troubleshoot the root cause of the failure
2021-10-13 21:05:26 [!]  AWS::EC2::SecurityGroup/SSH: DELETE_IN_PROGRESS
2021-10-13 21:05:26 [!]  AWS::IAM::Role/NodeInstanceRole: DELETE_IN_PROGRESS
2021-10-13 21:05:26 [!]  AWS::EC2::LaunchTemplate/LaunchTemplate: DELETE_IN_PROGRESS
2021-10-13 21:05:26 [!]  AWS::IAM::Policy/PolicyAWSLoadBalancerController: DELETE_IN_PROGRESS
2021-10-13 21:05:26 [✖]  AWS::IAM::Policy/PolicyAWSLoadBalancerController: CREATE_FAILED – "Resource creation cancelled"
2021-10-13 21:05:26 [✖]  AWS::EKS::Nodegroup/ManagedNodeGroup: CREATE_FAILED – "The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI. (Service: AmazonEKS; Status Code: 400; Error Code: InvalidRequestException; Request ID: 0eee90df-2bb0-48d7-b622-6bf3221e02a2; Proxy: null)"
2021-10-13 21:05:26 [ℹ]  1 error(s) occurred and nodegroups haven't been created properly, you may wish to check CloudFormation console
2021-10-13 21:05:26 [ℹ]  to cleanup resources, run 'eksctl delete nodegroup --region=us-east-1 --cluster=k8s --name=<name>' for each of the failed nodegroup
2021-10-13 21:05:26 [✖]  waiting for CloudFormation stack "eksctl-k8s-nodegroup-k8sx": ResourceNotReady: failed waiting for successful resource state
Error: failed to create nodegroups for cluster "k8s"

**Comment by ecc256** [comment]
Does it depend on eksctl version?

**Comment by ctalledo** [comment]
> Does it depend on eksctl version?

I don't believe so; it was working with eksctl v0.59.0 about a month ago, but now it does not. 

I upgraded eksctl to 0.70.0 today, but it did not resolve the problem either.

Thanks for the help!

**Comment by cPu1** [comment]
> > I'm seeing following behavior:
> > --instance-selector-cpu-architecture x86_64 does select arm64 image.
> > --instance-selector-cpu-architecture arm64 cannot find any images.
> 
> ARM support was missing for Ubuntu nodegroups. Both issues have been addressed in #4367. The fix will be out in the next release.

This is out in https://github.com/weaveworks/eksctl/releases/tag/v0.72.0-rc.0. Please try it out and feel free to reopen if you're still facing issues.
